content/validation/before/domains.md

```diff
@@ -0,0 +1,39 @@
+> Domains of training (used to train the model) and inference (used to make final predictions) data sets should not sizeably diverge.
+
+??? example "Examples"
+    * In most of the cases the model is usually trained on MC simulated data and later on applied to data to produce predictions which are then passed on to statistical inference step. MC simulation isn't perfect and therefore there are always differences between simulation and data domains. This can lead to the cases when model learns simulation artefacts which come e.g. from detector response mismodelling. Thus, its performance on data may be at least suboptimal and at most meaningless.
+    * Consider the model which is trained to predict the energy of a hadron given its energy deposits in the calorimeter (represented e.g. in the form of image or graph). Data consists of the showers initiated by a particle generated by a particle gun and having discrete values of energies (e.g. 1 GeV, 10 GeV, 20 GeV, etc.). However, in the real world settings, the model will be applied to showers produced by particles with underlying continuous energy spectrum. Although ML models are known for their capability to interpolate beyond their training domain, without apropriate tests model performance in the parts of the energy spectrum outside of its training domain is not a priori clear.  
+
+### Solution    
+
+It is a widely known practice to introduce **scale factor** (SF) **corrections** to account for possible discrepancies between data and MC simulation. Effectively, that means that the model is probed on some part of the domain on which it wasn't trained on (data) and then corrected for any differences by using a meaningful set of observables to derive SFs. One particularly promising approaches to remedy for data/MC domain difference is to use adversarial approaches to fully leverage the multidimensionality of the problem, as described in a [DeepSF note](https://cds.cern.ch/record/2666647).
+
+Another solution would be to incorporate methods of [**domain adaptation**](https://github.com/zhaoxin94/awesome-domain-adaptation) into an ML pipeline, which essentially guide the model to be invariant and robust towards domain shift. Particularly in HEP, a [Learning to Pivot with Adversarial Networks](https://arxiv.org/pdf/1611.01046.pdf) paper was one of the pioneers to investigate how a pile-up dependency can be mitigated, which can also be easily expanded to building a model robust to domain shift[^1].  
+
+Last but not the least, a usage of [**bayesian neural networks**]() has a great advantage of getting uncertainties estimate along with each prediction. If these uncertainties are significantly larger for some samples, this could indicate that they come from the domain beyond the training one (a so-called [out-of-distribution samples](https://ai.googleblog.com/2019/12/improving-out-of-distribution-detection.html)) - and therefore care needs to be taken in that cases.
```

Comment:
can you link to the documentation section on bayesian neural networks here? and maybe provide a link to a paper or example that describes what kind of care would need to be taken?